### PR TITLE
Support for session refresh

### PR DESCRIPTION
**Pull request by tenyo**
- moved session initialization logic to orchestrator
- added support for refreshing the session for existing client connections

### CODE REVIEW

File: api/orchestrators.go [hunk 1]
```diff
@@ -1,29 +1,73 @@
...
+// newDocDBOrchestrator creates a new session and initializes all clients
+func (s *server) newDocDBOrchestrator(ctx context.Context, sp *sessionParams) (*docDBOrchestrator, error) {
+	log.Debug("initializing docDBOrchestrator")
+
+	sess, err := s.assumeRole(
+		ctx,
+		s.session.ExternalID,
+		sp.role,
+		sp.inlinePolicy,
+		sp.policyArns...,
```
Comment (tenyo) [inline api/orchestrators.go:38]: We could update `assumeRole` to just take a `sessionParam` here, but I just left it for now

File: api/orchestration.go [hunk 2]
```diff
@@ -90,8 +90,13 @@ func (o *docDBOrchestrator) documentDBCreate(ctx context.Context, req *DocDBCrea
 		if err = retry(10, 3, 10*time.Second, func() error {
 			msgChan <- fmt.Sprintf("checking if docdb cluster %s is available before continuing", cl)
 
+			if err := o.refreshSession(ctx); err != nil {
+				msgChan <- fmt.Sprintf("unable to refresh orchestrator session: %s", err)
+				return err
```
Comment (fishnix) [inline api/orchestration.go:95]: I'm noticing that all of these write to the `msgChan` before returning `err` which writes to `errChan` (which also puts a message on the flywheel log). I don't think it's necessary.

**Review by fishnix — APPROVED** [review]
This is really nice, I like having the orchestrator generate the session and the refreshSession call.  I didn't think about having the `sessionParams` struct, but I like it!

We should think about creating an automatically closing background routine that does this for every orchestrator...  it could be as simple as a goroutine that has a ticker and a cancel channel and maybe a mux or a channel for synchronization... then the orchestrator doesn't need to think about the refresh at all! 🎉

File: api/orchestration.go (hunk 2 above)
Comment (tenyo) [reply api/orchestration.go:95]: hmm, i was looking at this and i'm not sure if it's correct, so we still may want to have the messages.
isn't only the last error written to `errChan` (when we time out)?

```
  errChan <- fmt.Errorf("failed to create docdb cluster %s, timeout waiting to become available: %s", cl, err.Error())
```

Comment (fishnix) [reply api/orchestration.go:95]: ah -- yeah that's right - but do we want to see those in the log?  i guess it doesn't hurt.

File: api/orchestration.go [hunk 3]
```diff
@@ -90,8 +90,13 @@ func (o *docDBOrchestrator) documentDBCreate(ctx context.Context, req *DocDBCrea
 		if err = retry(10, 3, 10*time.Second, func() error {
 			msgChan <- fmt.Sprintf("checking if docdb cluster %s is available before continuing", cl)
 
+			if err := o.refreshSession(ctx); err != nil {
```
Comment (fishnix) [inline api/orchestration.go]: this should be the task context ... sorry I missed it earlier 😆

**Pull request merged**